shared/src/commonMain/kotlin/com/baijum/ukufretboard/data/sync/BackupCodec.kt

```
@@ -0,0 +1,432 @@
+        )
+
+        var modified = false
+        val out = buildJsonObject {
+            for ((key, value) in root) {
+                val arrTsFields = arrayFields[key]
+                val objTsFields = objectFields[key]
+                when {
+                    arrTsFields != null -> {
+                        val arr = value as? JsonArray
+                        if (arr != null) {
+                            put(key, normalizeTimestampArray(arr, *arrTsFields))
```

detekt (1.23.8)

[warning] In most cases using a spread operator causes a full copy of the array to be created before calling a method. This may result in a performance penalty.

(detekt.performance.SpreadOperator)